pymc_marketing/mmm/transformers.py

```diff
@@ -43,22 +56,40 @@ def batched_convolution(x, w, axis: int = 0):
     # The last dimension of x is the "time" axis, which doesn't get broadcast
     # The last dimension of w is the number of time steps that go into the convolution
     x_shape, w_shape = params_broadcast_shapes([x.shape, w.shape], [1, 1])
+
     x = pt.broadcast_to(x, x_shape)
     w = pt.broadcast_to(w, w_shape)
     x_time = x.shape[-1]
-    shape = (*x.shape, w.shape[-1])
     # Make a tensor with x at the different time lags needed for the convolution
+    x_shape = list(x.shape)
+    # Add the size of the kernel to the time axis
+    x_shape[-1] = x_shape[-1] + w.shape[-1] - 1
+    shape = [*x_shape, w.shape[-1]]
```

Comment:
More/less readable?
```suggestion
    shape = (*x_shape[:-1], x_shape[-1] + w.shape[-1] - 1, w.shape[-1]]
```

Reply:
My issue was with the intermediate variable `x_shape` that is changed in place and no longer holds the shape of x. 
If someone, like me, starts by reading the line `shape = [...]` there's a second of confusion.